test/filestorage.js

```diff
@@ -0,0 +1,334 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { create } = require('..');
+const utils = require('../lib/utils');
+const metatests = require('metatests');
+const common = require('metarhia-common');
+
+const testDir = path.join(__dirname, 'fs-root');
+const minCompressSize = 1000;
+
+const testData = [
+  [
+    'data0',
+    '20cd1c30',
+    '4285d10832a8edf4cc7979e9a257e145dc5c934549f62e0bf1dc6070e67cc4ab',
+  ],
+  [
+    'data1',
+    '57ca2ca6',
+    '5b41362bc82b7f3d56edc5a306db22105707d01ff4819e26faef9724a2d406c9',
+  ],
+  [
+    'data2',
+  ],
+  [
+    'data3',
+  ],
+  [
+    'data4',
+  ],
+  [
+    'data5',
+  ],
+  [
+    'data6'.repeat(1000),
+  ],
+  [
+    'data7'.repeat(1000),
+  ],
+  [
+    'data8',
+    '2e169402',
+    'b5cc74ab5bb5a5f1acc7407be3e4cbce8611c5ed07354ab9e510b74ee0b273cb',
+  ],
+  [
+    'data9',
+    '5911a494',
+    'bbe0aa41024faeac81813a0194a95637d54cc65c025e0efd857ce0afcd51573f',
+  ],
+  [
+    'data10'.repeat(1000),
+    '828aaf45',
+    '7a8553c5934fd3b4a068a3d4c5bafc189e1fe8e0f7f46cbcc0fc1199249a39a2',
+  ],
+  [
+    'data11'.repeat(1000),
+    '5928f9a',
+    'f1cf91dea01f77fd860645de51462794cfbbae0a14d81350b21934e58a69941d',
+  ],
+  [
+    '白い猫でも黒い猫でも鼠を取る猫はいい猫だ',
+    '289722d',
+    'fab270bfcdc81e464611cd112e95c6b9590ba55483fbfe2dda98d67ddc741ab3',
+  ],
+];
+
+const fsTest = metatests.test('Filestorage');
+fsTest.endAfterSubtests();
+
+fsTest.beforeEach((test, cb) => {
+  create({ dir: testDir, minCompressSize }, (err, storage) => {
+    test.error(err);
+    cb({ storage });
+  });
+});
+
+fsTest.afterEach((test, cb) => {
+  utils.cleanup(testDir);
+  cb();
+});
+
+fsTest.test('Create root directory on create', test => {
+  const dir = testDir + 'create';
+  create({ dir }, err => {
+    test.error(err);
+    test.strictSame(fs.existsSync(dir), true);
+    utils.cleanup(dir);
+    test.end();
+  });
+});
+
+fsTest.test('Write string to file', (test, { storage }) => {
+  const data = testData[0];
```

Comment:
Also, what's the use of `testData` array? Why don't you just write that down at the beginning of te appropriate tests?